src/Functions/FunctionsMergeTreePartInfo.cpp

```diff
@@ -27,27 +27,101 @@ namespace ErrorCodes
 namespace
 {
 
+struct UnpackedPartSegments
+{
+    MergeTreePartInfo part_info;
+    std::string prefix;
+    std::string suffix;
+};
+
+std::optional<MergeTreePartInfo> tryParseMergeTreePartInfo(std::string_view data)
+{
+    return MergeTreePartInfo::tryParsePartName(data, MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING);
+}
+
 MergeTreePartInfo constructPartInfo(std::string_view data)
 {
-    if (auto info = MergeTreePartInfo::tryParsePartName(data, MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING))
+    if (auto info = tryParseMergeTreePartInfo(data))
         return std::move(info.value());
 
     throw Exception(ErrorCodes::BAD_ARGUMENTS, "Invalid MergeTree Part Name.");
 }
 
+/// Tries to parse part name format: "<prefix>_<part_name>_<tryN>".
+UnpackedPartSegments unpackPartName(std::string_view data)
+{
+    UnpackedPartSegments unpacked;
+
+    /// prefix_name_tryN
+    ///                 ^
+    size_t right = data.size();
+    size_t first_delimiter = data.find_first_of('_');
+    size_t last_delimiter = data.find_last_of('_');
+    size_t number_of_segments = std::ranges::count(data, '_') + 1;
+
+    if (number_of_segments < 4)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Invalid MergeTree Part Name: '{}'", data);
+
+    /// Process suffix first because it can be easily determined.
+    if (data.substr(last_delimiter + 1).starts_with("try"))
+    {
+        unpacked.suffix = data.substr(last_delimiter + 1);
+        right = last_delimiter;
+        number_of_segments -= 1;
+    }
+
+    switch (number_of_segments)
+    {
+    case 6: /// prefix_partition_min_max_level_mutation
+    {
+        unpacked.prefix = data.substr(0, first_delimiter);
+        unpacked.part_info = constructPartInfo(data.substr(first_delimiter + 1, right - (first_delimiter + 1)));
+        break;
+    }
+    case 5: /// prefix_partition_min_max_level or partition_min_max_level_mutation
+    {
+        /// Still incorrect if partition can be parsed as number and will be < min
+        if (auto info = tryParseMergeTreePartInfo(data.substr(0, right)))
+        {
+            unpacked.prefix = "";
+            unpacked.part_info = std::move(info.value());
+            break;
+        }
+
+        if (auto info = tryParseMergeTreePartInfo(data.substr(first_delimiter + 1, right - (first_delimiter + 1))))
+        {
+            unpacked.prefix = data.substr(0, first_delimiter);
+            unpacked.part_info = std::move(info.value());
+            break;
+        }
+
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Invalid MergeTree Part Name: '{}'", data);
```

Comment:
```suggestion
        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Invalid MergeTree part name: '{}'", data);
```
Capital latter seems strange here, isn't it?